assets/sip-big_vector/big_vector.move

```diff
@@ -0,0 +1,339 @@
+    // ======== Functions ========
+
+    /// create BigVector
+    public fun new<Element: store>(slice_size: u64, ctx: &mut TxContext): BigVector<Element> {
+        let id = object::new(ctx);
+        let slice_count = 1;
+        dynamic_field::add(&mut id, slice_count, vector::empty<Element>());
+        BigVector<Element> {
+            id,
+            slice_count,
+            slice_size,
+            length: 0,
+        }
+    }
+
+    /// return the slice_count of the BigVector
+    public fun slice_count<Element: store>(bv: &BigVector<Element>): u64 {
+        bv.slice_count
+    }
+
+    /// return the max size of a slice in the BigVector
+    public fun slice_size<Element: store>(bv: &BigVector<Element>): u64 {
+        bv.slice_size
+    }
+
+    /// return the size of the BigVector
+    public fun length<Element: store>(bv: &BigVector<Element>): u64 {
+        bv.length
+    }
+
+    /// return the slice_id related to the index i
+    public fun slice_id<Element: store>(bv: &BigVector<Element>, i: u64): u64 {
+        (i / bv.slice_size) + 1
+    }
+
+    /// return true if the BigVector is empty
+    public fun is_empty<Element: store>(bv: &BigVector<Element>): bool {
+        bv.length == 0
+    }
+
+    /// push a new element at the end of the BigVector
+    public fun push_back<Element: store>(bv: &mut BigVector<Element>, element: Element) {
+        if (length(bv) / bv.slice_size == bv.slice_count) {
+            bv.slice_count = bv.slice_count + 1;
+            let new_slice = vector::singleton(element);
+            dynamic_field::add(&mut bv.id, bv.slice_count, new_slice);
+        }
+        else {
+            let slice = dynamic_field::borrow_mut(&mut bv.id, bv.slice_count);
+            vector::push_back(slice, element);
+        };
+        bv.length = bv.length + 1;
+    }
+
+    /// pop an element from the end of the BigVector
+    public fun pop_back<Element: store>(bv: &mut BigVector<Element>): Element {
+        let slice = dynamic_field::borrow_mut(&mut bv.id, bv.slice_count);
+        let element = vector::pop_back(slice);
+        trim_slice(bv);
+        bv.length = bv.length - 1;
+
+        element
+    }
+
+    /// borrow an element at index i from the BigVector
+    public fun borrow<Element: store>(bv: &BigVector<Element>, i: u64): &Element {
+        let slice_count = (i / bv.slice_size) + 1;
+        let slice = dynamic_field::borrow(&bv.id, slice_count);
+        vector::borrow(slice, i % bv.slice_size)
+    }
+
+    /// borrow a mutable element at index i from the BigVector
+    public fun borrow_mut<Element: store>(bv: &mut BigVector<Element>, i: u64): &mut Element {
+        let slice_count = (i / bv.slice_size) + 1;
+        let slice = dynamic_field::borrow_mut(&mut bv.id, slice_count);
+        vector::borrow_mut(slice, i % bv.slice_size)
+    }
+
+    /// borrow a slice from the BigVector
+    public fun borrow_slice<Element: store>(bv: &BigVector<Element>, slice_count: u64): &vector<Element> {
```

Comment:
I can see this being helpful but if we remove the one below I am not sure it needs to exist.
Have you found this to be an important API?